lib/zendesk_api/error.rb

```diff
@@ -2,16 +2,20 @@ module ZendeskAPI
   module Error
     class ClientError < Faraday::Error::ClientError
       attr_reader :wrapped_exception
+
+      def to_s
+        "#{super} -- #{@response.method} #{@response.url}"
+      end
     end
 
     class RecordInvalid < ClientError
       attr_accessor :response, :errors
 
-      def initialize(response)
-        @response = response
+      def initialize(*)
+        super
 
-        if response[:body].is_a?(Hash)
-          @errors = response[:body]["details"] || response[:body]["description"]
+        if @response[:body].is_a?(Hash)
```

Comment:
There's still an attr accessor for the response. You don't need to change to using ivars.